qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -40,7 +42,7 @@ def __init__(
         self,
         quantum_instance: QuantumInstance,
         optimizer: Optional[Optimizer] = None,
-        initial_point: Optional[np.ndarray] = None,
+        initial_point: Optional[Union[np.ndarray, InitialPoint]] = None,
```

Comment:
I am wondering if we should add a `HFInitialPoint` class which simply returns an all zero array of the correct length here.
That way we could have this as a default. This would be a very minimal class obviously but I think it could be a bit clearer because users may not necessarily know that the all-zero case is simply the HF initial point.

Just putting this out there. Also tagging @woodsp-ibm and @pbark

Reply:
I agree with @mrossinek that this would be a good addition. Idk if this should be done here or in separate PR. @declanmillar if you can add this functionality also would be great.

Reply:
Yep, I think that sounds like a usefully illustrative addition. I'm travelling over the next few days, but I will chip away to resolve these comments when stuck on trains and planes.

Reply:
Having an HFInitialPoint sounds fine - it would go some way to sorting out #571 , in regards of removing the preferred initial point logic from EvolvedOperatorAnsatz. Something is needed for UVCC Factory though to set initial point to 0 once that is removed as a preferred initial point that VQE gets from the EvolvedOperatorAnsatz

Reply:
I decided to open a separate PR for this (#632) for more clarity.